src/plugins/data_view_field_editor/public/components/preview/field_preview_context.tsx

```diff
@@ -288,7 +288,7 @@ export const FieldPreviewProvider: FunctionComponent<{ controller: PreviewContro
       fields: fields.map((field) => {
         const nextValue =
           script === null && Boolean(document)
-            ? get(document?._source, name ?? '') ?? get(document?.fields, name ?? '') // When there is no script we try to read the value from _source/fields
+            ? get(document?.fields, name ?? '') ?? get(document?._source, name ?? '') // When there is no script we try to read the value from _source/fields
```

Comment:
The runtime field still gets the value from _source if there's no script. The change is showing indexed values in the preview pane, there's no change in runtime field behavior.

Reply:
```suggestion
            ? get(document?._source, name ?? '')
```

If there is no script and the target field is unmapped, `get(document?.fields, name ?? '')` would be empty anyway, so this condition is redundant.

If there is no script and the target field is mapped, reading the value from `fields` here would be against the description:
<img width="500" alt="Screenshot 2024-04-22 at 13 36 35" src="https://github.com/elastic/kibana/assets/1415710/ec7c4f90-e8e5-43c8-b8ba-21ba1821071e">

Reply:
ah, I think I see now. Did it ever make sense to get the value from fields? It didn't as best I can tell.

Reply:
Your change caused a failure with tests introduced with this PR - which gives a good explanation of why you might get a value from a field that doesn't exist in source - https://github.com/elastic/kibana/pull/145943

Reply:
Why not to keep what we have in main then? First in source, then in fields as fallback.

Reply:
@jughosta Perhaps we should discuss this with the team. I don't think there's one right answer to this.

Reply:
I'm also a bit confused here tbh and would like to understand this change before merging. My understanding based on our discussion yesterday is this:
- We will show the values from `fields` for sample entries in the preview list.
- Specifically for the preview of the field currently being edited:
  - If it is not a runtime field (e.g. adding a formatter to an existing field), we will show the `fields` value.
  - If it is a runtime field without a script, we will show the `_source` value.
  - If it is a runtime field with a script, we will show the value the script returns.

Does this seem correct? If so, it looks like the change in this PR does not align with our decision since it will now try to pull from `fields` first for runtime fields without a script. Based on my understanding, I think @jughosta is correct that we should instead use `get(document?._source, name ?? '')` without a fallback in order to emulate the same logic ES uses when handling runtime fields without a script.